app/controllers/my_facilities/drug_stocks_controller.rb

```diff
@@ -59,9 +59,17 @@ def create_drug_report
       .includes(facility_group: :protocol_drugs)
       .where(protocol_drugs: {stock_tracked: true})
 
+    @blocks = Region
+      .block_regions
+      .joins("INNER JOIN regions facility_region ON regions.path @> facility_region.path")
+      .where(facility_region: {source_id: @facilities})
+      .distinct("regions.id")
+
     @for_end_of_month_display = @for_end_of_month.strftime("%b-%Y")
-    render && return if @facilities.empty?
-    @query = DrugStocksQuery.new(facilities: @facilities, for_end_of_month: @for_end_of_month)
+    @query = DrugStocksQuery.new(facility_group: @selected_facility_group,
+                                 facilities: @facilities,
+                                 blocks: @blocks,
```

Comment:
Why do we need to pass blocks in here? Can't the `DrugStocksQuery` figure this out? If it's only to expose it to the view, I'd rather do something like `@blocks = @query.blocks`. I think it's better to avoid room for inconsistent input.